extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -46,37 +43,39 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
      * @param {Object} entry - The record log entry from metadata.
      * @return {undefined}
      */
-    _updateZookeeper(entry) {
+    _updateZkBucketNode(entry) {
         const value = JSON.parse(entry.value);
-        const path = `${lifecycleZkPath}/${value.owner}:${value.name}`;
-        return this.zkClient.exists(path, (err, stat) => {
+        const { zookeeperPath } = this.extConfig;
+        const path =
+            `${zookeeperPath}/data/buckets/${value.owner}:${value.name}`;
+        // Remove existing node if deleting the bucket or its configuration.
+        if (value.deleted || value.lifecycleConfiguration === null) {
+            return this.zkClient.remove(path, err => {
+                if (err && err.name !== 'NO_NODE') {
+                    this.log.error('Could not remove zookeeper node', {
+                        method: 'LifecycleQueuePopulator._updateZkBucketNode',
+                        zkPath: path,
+                        error: err,
+                    });
+                }
+                return undefined;
+            });
+        }
+        return this.zkClient.exists(path, err => {
```

Comment:
I think you don't need this call (actually you don't use the result at all, passed as second argument), you can just do the `create` directly.

You'll also need to check for `NODE_EXISTS` error from `create` and don't log an error in such case, because it's a case that can happen in normal behavior (a debug log is advisable though).

Reply:
True. I will update this.